src/Microsoft.Dnx.Tooling/List/MismatchDependencyRenderer.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Dnx.Tooling.Algorithms;
+using NuGet;
+
+namespace Microsoft.Dnx.Tooling.List
+{
+    internal class MismatchDependencyRenderer
+    {
+        private readonly DependencyListOptions _options;
+        private readonly FrameworkName _framework;
+
+        public MismatchDependencyRenderer(DependencyListOptions options, FrameworkName framework)
+        {
+            _options = options;
+            _framework = framework;
+        }
+
+        public void Render(IGraphNode<LibraryDependency> root)
+        {
+            // tuples of <Library Name, Requested Version, Actual Version>
+            var results = new HashSet<Tuple<string, string, string>>();
```

Comment:
Tuple helps a lot. To write a private class I need to further override `Equals` and `GetHashCode` this is much more simpler. The data is not further reused. It just represents three columns.